src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java

```diff
@@ -217,7 +226,74 @@ private void validateMedicalInventory(MedicalInventory medInventory) throws OHDa
 			throw new OHDataValidationException(errors);
 		}
 	}
-
+	
+	/**
+	 * Validate the Inventory rows of inventory.
+	 *
+	 * @param inventory - The  {@link MedicalInventory}
+	 * @param inventoryRowSearchList- The list of {@link MedicalInventory}
+	 * @throws OHDataValidationException
+	 */
+	@Transactional(rollbackFor = OHServiceException.class)
+	public void validateInventory(MedicalInventory inventory, List<MedicalInventoryRow> inventoryRowSearchList) throws OHDataValidationException, OHServiceException {
+		LocalDateTime movFrom = inventory.getLastModifiedDate();
+		LocalDateTime movTo = TimeTools.getNow();
+		String medDescriptionForUpdate = "";
+		String medDescriptionForNew = "";
+		List<Movement> movements = movBrowserManager.getMovements(null, null, null, null, movFrom, movTo, null, null, null, null); 
+		boolean updated = false;
+		boolean added = false;
+		if (!movements.isEmpty()) {
+			Map<Medical, List<Movement>> groupedByMedical = movements.stream().collect(Collectors.groupingBy(Movement::getMedical));
+			for (Iterator<MedicalInventoryRow> iterator = inventoryRowSearchList.iterator(); iterator.hasNext();) {
+				MedicalInventoryRow medicalInventoryRow = (MedicalInventoryRow) iterator.next();
+				String lotCode = medicalInventoryRow.getLot().getCode();
+				double theoQty = medicalInventoryRow.getTheoreticQty();
+				Medical medical = medicalInventoryRow.getMedical();
+				List<Movement> movs = groupedByMedical.get(medical);
+				if (movs != null) {
+					Movement mov = movs.get(0);
+					String lotCodeOfMovement = mov.getLot().getCode();
+					List<Lot> lots = movStockInsertingManager.getLotByMedical(medical).stream().filter(l -> l.getCode().equals(lotCodeOfMovement)).collect(Collectors.toList());
+					if (lots != null) {
+						for (Lot lot : lots) {
+							double mainStoreQty = (double)lot.getMainStoreQuantity();
+							if (lotCodeOfMovement.equals(lotCode)) {
+								if (mainStoreQty != theoQty) { 
+									medicalInventoryRow.setTheoreticQty(mainStoreQty);
+									medicalInventoryRow = medicalInventoryRowManager.updateMedicalInventoryRow(medicalInventoryRow);
+									if (medicalInventoryRow != null) {
+										updated = true;
+										medDescriptionForUpdate = medDescriptionForUpdate + " "+medical.getDescription()+" lot code "+lot+" and qty :"+mainStoreQty+"/";
```

Comment:
Also, the last "/" is meant to be "\n" (new line)?

```suggestion
										medDescriptionForUpdate = medDescriptionForUpdate + " " + medical.getDescription() + " lot code " + lot + " and qty : " + mainStoreQty + "/";
```